### PR TITLE
doc: add upgrade instructions for v0.6.1 to v0.7.0 

### PR DESCRIPTION
**Pull request by fanminshi**
remove incompatible_upgrade.md since it is not used anywhere.
add upgrade instructions for v0.6.1 to v0.7.0.

[skip ci]

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hasbro17 @hongchaodeng

**Comment by hongchaodeng** [comment]
I think we still need to keep "incompatible upgrade guide".
It's just that we don't need the old backup guide.

File: doc/user/upgrade/upgrade_guide.md [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
 
-In the case of an upgrade failure you can restore your cluster to the previous state from the previous backup. See the [spec examples](https://github.com/coreos/etcd-operator/blob/master/doc/user/spec_examples.md#three-members-cluster-that-restores-from-previous-pv-backup) on how to do that.
+When upgrading operator from a version that's < v0.7.0, please [follow previous Backup safety precaution](https://github.com/coreos/etcd-operator/blob/v0.6.1/doc/user/upgrade/upgrade_guide.md#backup-safety-precaution)
 
+## 0.6.1 -> 0.7.0
+**Note:** If your cluster doesn't use backup policy and restore policy, then do the following:
 
-### Upgrade operator deployment
-An [in-place update](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#in-place-updates-of-resources) can be performed when the upgrade is compatible, i.e we can upgrade the operator without affecting the cluster.
+- Start to edit the existing etcd deployment: `kubectl edit deployment/<your-operator>`.
```
Comment (hasbro17) [inline doc/user/upgrade/upgrade_guide.md]: Edit the existing etcd-operator deployment

**Comment by fanminshi** [comment]
@hongchaodeng the issue with `incompatible upgrade guide` is that it is not relevant to `0.7.0 ` anymore. everything in the guide only applies to version that's < `0.7.0`. I took a look of previous version upgrade guide, I don't see any reference to `incompatible upgrade guide`.  cc/ @hasbro17

**Comment by fanminshi** [comment]
cc/ @hasbro17

**Comment by hasbro17** [comment]
@fanminshi Can you structure the `0.6.1` to `0.7.0` section to something like:

```
Note: If your cluster spec specifies either the backup policy or restore policy, then follow the migrate CR guide to update the cluster spec before upgrading the etcd-operator deployment.

Update the etcd-operator deployment:
...
```

**Comment by fanminshi** [comment]
all fixed. PTAL. cc/ @hasbro17

File: doc/user/upgrade/upgrade_guide.md [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+In the case of an upgrade failure you can restore your cluster to the previous state from the previous backup. See the [spec examples](https://github.com/coreos/etcd-operator/blob/v0.6.1/doc/user/spec_examples.md) on how to do that.
 
-### Upgrade operator deployment
-An [in-place update](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#in-place-updates-of-resources) can be performed when the upgrade is compatible, i.e we can upgrade the operator without affecting the cluster.
+## v0.6.1 -> v0.7.0
+**Note:** if you cluster specifies either the backup policy or restore policy, then follow the  [migrate CR](./migrate_cr_070.md) guide to update the cluster spec before upgrading the etcd-operator deployment.
 
-To upgrade an operator deployment the image field `spec.template.spec.containers.image` needs to be changed via an in-place update.
+Update the etcd-operator deployment:
+- Edit the existing etcd deployment: `kubectl edit deployment/<your-operator>`.
```
Comment (hasbro17) [inline doc/user/upgrade/upgrade_guide.md]: Edit the existing etcd-operator deployment:

**Comment by hasbro17** [comment]
LGTM after nit.

File: doc/user/upgrade/upgrade_guide.md (hunk 2 above)
Comment (fanminshi) [reply doc/user/upgrade/upgrade_guide.md]: fixed!

**Pull request merged**